fix: handle invalid addresses in useAddress hook

snome-aly opened the pull request:
---
  Summary

  This PR improves the handling of invalid addresses in the useAddress hook and Address component:

  - Optimize performance: Skip ENS queries when address is invalid to avoid unnecessary network requests
  - Better UX: Display clear error message with warning icon for invalid addresses instead of showing loading skeleton forever
  - Customizable: Support i18n through invalidAddressText prop and brand theming through invalidAddressColor prop
  - Updated docs: Add new props documentation and invalid address usage examples

  Changes

  useAddress Hook (packages/hooks/src/useAddress.ts)

  - Add enabled: !!checkSumAddress to ENS queries to prevent execution when address is invalid
  - Simplify logic by removing intermediate isValidAddress variable
  - Optimize computed values to only calculate when address is valid

  Address Component (packages/components/src/Address/Address.tsx)

  - Add invalid address error UI with warning icon and skeleton-like styling
  - New props:
    - invalidAddressText (default: "Invalid address") - for i18n support
    - invalidAddressColor (default: "#ef4444") - for brand theming
  - Three distinct states: loading skeleton, error display, and normal display
  - Uses hexToRgba helper to convert hex color to rgba for background opacity

  Documentation (docs/pages/components/Address.mdx)

  - Document new props in props table
  - Add "Invalid Address Handling" section with customization examples

  Test plan

  - Test with valid address - displays normally with ENS resolution
  - Test with invalid address - shows error message with warning icon
  - Test with no address - shows loading skeleton
  - Verify ENS queries are not executed for invalid addresses
  - Test custom error text for i18n
  - Test custom error color for branding

  ---

rin-st commented on docs/pages/components/Address.mdx:
````diff
@@ -148,3 +148,27 @@ import { Address } from "@scaffold-ui/components";
+<Address address="0xInvalidAddress123" />;
+```
+
+To customize the error color, override the CSS variable in your theme:
````
Same as https://github.com/scaffold-eth/scaffold-ui/pull/74/files#r2529705258, remove this part please

snome-aly replied:
done

rin-st commented on packages/hooks/src/useAddress.ts:
```diff
@@ -29,26 +29,33 @@ export const useAddress = (UseAddressOptions: UseAddressOptions) => {
+    ? getBlockExplorerAddressLink(UseAddressOptions?.chain ?? mainnet, checkSumAddress)
+    : "";
+
+  // Use checkSumAddress for blockie generation to ensure consistency
```
remove comment

rin-st commented on packages/components/src/Address/Address.tsx:
```diff
@@ -36,20 +36,68 @@ export const Address: React.FC<AddressProps> = ({
+        <div className="flex flex-col space-y-1">
+          <span
+            className={`ml-1.5 ${textSizeMap[ensSize]} font-bold`}
+            style={{ color: "var(--color-sui-error)" }}
```
remove style

rin-st commented on packages/components/src/Address/Address.tsx:
```diff
@@ -36,20 +36,68 @@ export const Address: React.FC<AddressProps> = ({
+          </span>
+          <span
+            className={`ml-1.5 ${textSizeMap[addressSize]} break-all`}
+            style={{ color: "var(--color-sui-error)", opacity: 0.7 }}
```
remove style

snome-aly replied:
done

rin-st commented on packages/components/src/Address/Address.tsx:
```diff
@@ -36,20 +36,68 @@ export const Address: React.FC<AddressProps> = ({
+  if (address && !isValidAddress) {
+    return (
+      <DefaultStylesWrapper
+        className="flex items-center text-sui-primary-content"
```
use `text-sui-error` here instead of `text-sui-primary-content` and remove all the text-color related styles inside the `DefaultStylesWrapper`

snome-aly replied:
done

snome-aly replied:
done

snome-aly commented on packages/hooks/src/useAddress.ts:
```diff
@@ -29,26 +29,33 @@ export const useAddress = (UseAddressOptions: UseAddressOptions) => {
-  );
+  // If checkSumAddress exists, it means the address is valid (already validated by isAddress check above)
+  // So we can directly use !!checkSumAddress instead of calling isAddress again
+  const isValidAddress = !!checkSumAddress;
```
done

snome-aly commented:
> Thank you for the updates!
> 
> Added some comments

 Thanks for the detailed review @rin-st! I've addressed all your feedback:

  - Simplified boolean conversions using `Boolean()` instead of `!!`
  - Removed redundant logic checks
  - Cleaned up comments
  - Simplified error UI (removed bg color and wrapper)
  - Used `currentColor` for better CSS inheritance
  - Moved CSS customization docs as suggested

snome-aly commented on packages/components/src/Address/Address.tsx:
```diff
@@ -36,20 +36,68 @@ export const Address: React.FC<AddressProps> = ({
+          style={{
+            width: (blockieSizeMap[blockieSize] * 24) / blockieSizeMap["base"],
+            height: (blockieSizeMap[blockieSize] * 24) / blockieSizeMap["base"],
+            backgroundColor: "color-mix(in srgb, var(--color-sui-error) 10%, transparent)",
```
I removed the svg wrapper div since it became redundant after removing the background color. The wrapper was only setting width/height which the SVG already has.

snome-aly commented on packages/hooks/src/useAddress.ts:
```diff
@@ -29,26 +29,33 @@ export const useAddress = (UseAddressOptions: UseAddressOptions) => {
     query: {
-      enabled: Boolean(ens),
+      // Only fetch avatar when we have both valid address and ENS name
+      enabled: !!checkSumAddress && Boolean(ens),
```
done

snome-aly commented on packages/components/src/Address/Address.tsx:
```diff
@@ -36,20 +36,68 @@ export const Address: React.FC<AddressProps> = ({
+            height={(blockieSizeMap[blockieSize] * 16) / blockieSizeMap["base"]}
+            viewBox="0 0 24 24"
+            fill="none"
+            stroke="var(--color-sui-error)"
```
done

snome-aly commented on packages/hooks/src/useAddress.ts:
```diff
@@ -44,11 +44,13 @@ export const useAddress = (UseAddressOptions: UseAddressOptions) => {
-    UseAddressOptions?.chain ?? mainnet,
-    checkSumAddress ?? "",
-  );
+  const isValidAddress = Boolean(checkSumAddress);
```
Moved it up 👍
